sdk/core/azure-core/tests/testserver_tests/coretestserver/coretestserver/test_routes/streams.py

```diff
@@ -67,4 +66,16 @@ def compressed_no_header():
 @streams_api.route('/compressed', methods=['GET'])
 def compressed():
     return Response(stream_compressed_header_error(), status=300, headers={"Content-Encoding": "gzip"})
+
+def stream_decompress_header():
+    with gzip.open('test.tar.gz', 'wb') as f:
+        f.write(b"test")
+
+    with open(os.path.join(os.path.abspath('test.tar.gz')), "rb") as fd:
+        yield fd.read()
```

Comment:
I would suggest `yield`ing the contents at least twice (and possibly more). The idea being that we make the receiver have to call `read` multiple times. Which will verify that decompression works as expected even when presented with boundaries that may not align with blocks.